src/reporter/tests/test_notify.py

```diff
@@ -695,3 +695,151 @@ def test_json_ld(service, notification):
+        "unitCode": "C62",
+        "observedAt": "2021-01-28T12:33:20.000Z"
+        }
+    }
+    url = '{}'.format(notify_url)
+    get_url = "{}/entities/urn:ngsi-ld:Device:filling001".format(
+        QL_URL)
+    url_new = '{}'.format(get_url)
+    r = requests.post(url, data=json.dumps(notification),
+                      headers=notify_header(service))
+    assert r.status_code == 200
+    # Give time for notification to be processed
+    time.sleep(SLEEP_TIME)
+    res_get = requests.get(url_new, headers=query_header(service))
+    assert res_get.status_code == 200
+    assert_equal_time_index_arrays([res_get.json()['index'][0]], ["2021-01-28T12:33:10.000Z"])
+    delete_entity_type(service, notification['data'][0]['type'])
+
+
+@pytest.mark.parametrize("service", services)
+def test_json_ld_modified_at_meta(service, notification):
+    # example json-ld entity with modifiedAt as metadata
+    notification['data'][0] = {
+        "id": "urn:ngsi-ld:Device:filling001",
+        "type": "FillingSensor",
+        "filling": {
+            "type": "Property",
+            "value": 0.94,
+            "unitCode": "C62",
+            "modifiedAt": "2021-01-28T12:33:10.000Z"
+        }
+    }
+    url = '{}'.format(notify_url)
+    get_url = "{}/entities/urn:ngsi-ld:Device:filling001".format(
+        QL_URL)
+    url_new = '{}'.format(get_url)
+    r = requests.post(url, data=json.dumps(notification),
+                      headers=notify_header(service))
+    assert r.status_code == 200
+    # Give time for notification to be processed
+    time.sleep(SLEEP_TIME)
+    res_get = requests.get(url_new, headers=query_header(service))
+    assert res_get.status_code == 200
+    assert_equal_time_index_arrays([res_get.json()['index'][0]], ["2021-01-28T12:33:10.000Z"])
+    delete_entity_type(service, notification['data'][0]['type'])
+
+
+@pytest.mark.parametrize("service", services)
+def test_json_ld_both_at_meta(service, notification):
+    # example json-ld entity with modifiedAt and observedAt as metadata
+    notification['data'][0] = {
+        "id": "urn:ngsi-ld:Device:filling001",
+        "type": "FillingSensor",
+        "filling": {
+            "type": "Property",
+            "value": 0.94,
+            "unitCode": "C62",
+            "observedAt": "2021-01-28T12:33:20.000Z",
+            "modifiedAt": "2021-01-28T12:33:22.000Z"
+        }
+    }
+    url = '{}'.format(notify_url)
+    get_url = "{}/entities/urn:ngsi-ld:Device:filling001".format(
+        QL_URL)
+    url_new = '{}'.format(get_url)
+    r = requests.post(url, data=json.dumps(notification),
+                      headers=notify_header(service))
+    assert r.status_code == 200
+    # Give time for notification to be processed
+    time.sleep(SLEEP_TIME)
+    res_get = requests.get(url_new, headers=query_header(service))
+    assert res_get.status_code == 200
+    assert_equal_time_index_arrays([res_get.json()['index'][0]], ["2021-01-28T12:33:10.000Z"])
```

Comment:
This doesn't seem to match the `observedAt` in line 766

Reply:
indeed...